package/MDAnalysis/analysis/contacts.py

```diff
@@ -773,3 +773,167 @@ def plot_qavg(self, filename=None, **kwargs):
+
+        $X$ is a conformation,
+        $r_{ij}(X)$ is the distance between atoms $i$ and $j$ in conformation $X$,
+        $r^0_{ij}$ is the distance from heavy atom i to j in the native state conformation,
+        $S$ is the set of all pairs of heavy atoms $(i,j)$ belonging to residues $\theta_i$ and $\theta_j$ such that $|\theta_i - \theta_j| > 3$ and $r^0_{i,} < 4.5 \unicode{x212B}$,
+        $\beta=5 \unicode{x212B}^{-1}$,
+        $\lambda=1.8$ for all-atom simulations
+
+    """    
+    # reference groups A and B from selection strings
+    refA, refB = ref.select_atoms(selA), ref.select_atoms(selB)
+
+    # 2D float array, reference distances (r0)
+    dref = distance_array(refA.positions, refB.positions)
+
+    # 2D bool array, select reference distances that are less than the cutoff radius
+    mask = dref < radius
+    #print("ref has {:d} contacts within {:.2f}".format(mask.sum(), radius))
+
+    # group A and B in a trajectory
+    grA, grB = u.select_atoms(selA), u.select_atoms(selB)
+    results = []
+
+
+    for ts in u.trajectory:
+        d = distance_array(grA.positions, grB.positions)
+        r, r0 = d[mask], dref[mask]
+        x = 1/(1 + np.exp(beta*(r - lambda_constant * r0)))
+
+        # average/normalize and append to results
+        results.append(( ts.time, x.sum()/mask.sum() ))
+
+    #results = pd.DataFrame(results, columns=["Time (ps)", "Q"])
+    return results
+
+from MDAnalysis.analysis.distances import distance_array
+from .base import AnalysisBase
+logger = logging.getLogger(__name__)
+
+class BestHummerContacts(AnalysisBase):
+    r"""Calculate Best-Hummer fraction of native contacts (Q) from a atrajectory
+
+    .. versionadded:: 0.13.0
+    """
+    def __init__(self, grA, grB, refA, refB, radius=4.5, lambda_constant=1.8, beta=5.0,
+                 start=None, stop=None, step=None):
+        """Calculate the persistence length for polymer chains
+
+        Parameters
+        ----------
+        u : Universe
+            the thing with frame
+        grA, grB: AtomGroup
+            two contacting groups that change over time
+        refA, refB: AtomGroup
+            two contacting groups in their reference conformation
+        radius: float, optional (4.5 Angstroms)
+            radius within which contacts exist
+        lambda_constant: float, optional (1.8 unitless)
+            contact is considered formed between (lambda*r0,r0)
+        beta: float, optional (5 Angstroms^-1)
+            softness of the switching function, the lower the softer
+        start : int, optional
+            First frame of trajectory to analyse, Default: 0
+        stop : int, optional
+            Last frame of trajectory to analyse, Default: -1
+        step : int, optional
+            Step between frames to analyse, Default: 1
+
+
+        Attributes
+        ----------
+        results: list 
+            Fraction of native contacts for each frame
+        """
+        assert(grA.universe == grB.universe)
```

Comment:
You don't want an assert here, do a proper if not x==y: raise ValueError with a nice message